### ISSUE TITLE
Carthage fails when home directory is not on startup volume

### ISSUE THREAD

**Issue by simX** [Carthage/Carthage#711]
I recently upgraded to OS X 10.10, and I'm using Carthage 0.8.0, but my home directory is on my old partition.  So my home folder is located at `/Volumes/DifferentPartition/Users/simx/` .  My git repository is also on this other partition, as in `/Volumes/DifferentPartition/Users/simx/Development/SekritProject/`

In such a situation, carthage eventually fails during the downloading phase.  Here's the output of `carthage update --verbose`:

*** Cloning KIF
*** Cloning HockeySDK-iOS
*** Cloning stripe-ios
*** Cloning libPhoneNumber-iOS
*** Cloning ios-i18n
*** Cloning mixpanel-iphone
*** Cloning objc-TimesSquare
*** Cloning sskeychain
*** Cloning AFNetworking
*** Checking out stripe-ios at "v5.1.2"
*** Checking out mixpanel-iphone at "3c77e98c93422f99cc3506ddd23669691e6b5e03"
*** Downloading sskeychain at "Version 1.2.3"
*** Checking out libPhoneNumber-iOS at "0.8.7"
*** Checking out objc-TimesSquare at "978628c39fabc6c0c060536782f87a6c84012fb1"
*** Checking out ios-i18n at "v1.0.8"
*** Checking out AFNetworking at "eedc564b46cde9ae285d60fc393016f578258169"
*** Downloading HockeySDK-iOS at "Version 3.7.2"
*** Checking out HockeySDK-iOS at "3.7.2"
*** Checking out KIF at "v3.0.8"
*** Downloading sskeychain at "Version 1.2.3"
Error Domain=NSPOSIXErrorDomain Code=18 "The operation couldn’t be completed. Cross-device link"

**Comment by simX** [comment]
Hm, maybe false alarm, this seems related to the soffes/sskeychain repo

**Comment by ikesyo** [comment]
This is dupe of #706.

**Comment by ikesyo** [comment]
Closed by #713.

**Comment by pomozoff** [comment]
I've faced this issue:

```
PhaseScriptExecution Copy\ Carthage\ Dependencies /Volumes/Ramdisk/xudBgLqs/0/forbin/corp/software/hms/hms-ios-app/build/Build/Intermediates.noindex/ArchiveIntermediates/AppCenterDebug/IntermediateBuildFilesPath/App.build/AppCenterDebug-iphoneos/App.build/Script-4889E8AB1FF510D300A7AF56.sh (in target 'App' from project 'App')
    cd /Volumes/Ramdisk/xudBgLqs/0/forbin/corp/software/hms/hms-ios-app
    /bin/sh -c /Volumes/Ramdisk/xudBgLqs/0/forbin/corp/software/hms/hms-ios-app/build/Build/Intermediates.noindex/ArchiveIntermediates/AppCenterDebug/IntermediateBuildFilesPath/App.build/AppCenterDebug-iphoneos/App.build/Script-4889E8AB1FF510D300A7AF56.sh
+ /usr/local/bin/carthage copy-frameworks
replaceItem failed: Error Domain=NSCocoaErrorDomain Code=512 "The file “Core.framework” couldn’t be saved in the folder “Frameworks”." UserInfo={NSFileOriginalItemLocationKey=file:///Volumes/Ramdisk/xudBgLqs/0/forbin/corp/software/hms/hms-ios-app/build/Build/Intermediates.noindex/ArchiveIntermediates/AppCenterDebug/BuildProductsPath/AppCenterDebug-iphoneos/App.app/Frameworks/Core.framework/,
 NSURL=file:///Volumes/Ramdisk/xudBgLqs/0/forbin/corp/software/hms/hms-ios-app/build/Build/Intermediates.noindex/ArchiveIntermediates/AppCenterDebug/BuildProductsPath/AppCenterDebug-iphoneos/HomeMonitoring.app/Frameworks/Core.framework/, NSFileNewItemLocationKey=file:///var/folders/kn/x7l3ktg534nd935m4vzshcv40000gq/T/carthage-lipo-M2pUvb/Core.framework, NSUnderlyingError=0x7f9471fdb640 {Error Domain=NSPOSIXErrorDomain Code=18 "Cross-device link" UserInfo={NSURL=file:///Volumes/Ramdisk/xudBgLqs/0/forbin/corp/software/hms/hms-ios-app/build/Build/Intermediates.noindex/ArchiveIntermediates/AppCenterDebug/BuildProductsPath/AppCenterDebug-iphoneos/App.app/Frameworks/Core.framework/}}}
Command PhaseScriptExecution failed with a nonzero exit code
```
